tests/integration/devfile/cmd_dev_test.go

```diff
@@ -210,17 +210,17 @@ var _ = Describe("odo dev command tests", func() {
 			})
 
 			It("should expose the endpoint on localhost", func() {
-				helper.RunDevMode(func(session *gexec.Session, outContents, errContents []byte) {
-					i := strings.Index(string(outContents), "localhost")
-					url := fmt.Sprintf("http://%s", string(outContents)[i:i+15])
+				err := helper.RunDevMode(func(session *gexec.Session, outContents, errContents []byte) {
+					urls := helper.FindAllMatchingStrings(string(outContents), "localhost")
```

Comment:
Also, I was thinking about getting these URLs passed as parameters to the callback (with session, outContents, errcontents), as this will be something useful for a lot of tests, instead of repeating this code in every test. We could also get them as return value of StartDevMode()

Reply:
Makes sense. PTAL at https://github.com/redhat-developer/odo/pull/5526/commits/95b7a0c62e5a28949632ad95f26972c0b2b91a79.